src/wallet/wallet2.h

```diff
@@ -1850,6 +1956,11 @@ namespace tools
     uint64_t m_ignore_outputs_above;
     uint64_t m_ignore_outputs_below;
     bool m_track_uses;
+    bool m_is_background_wallet;
+    BackgroundSyncType m_background_sync_type;
+    bool m_background_syncing;
```

Comment:
I think the best way to get thread safe access to `m_background_syncing` is to let the caller handle thread safety, which I made sure to do via `LOCK_REFRESH` in the wallet API, using `LOCK_IDLE_SCOPE` in the CLI, and the RPC should be ok because the server runs in a single thread.

I think if we wanted to add more logic into wallet2 that handles thread safety surrounding `m_background_syncing`, then every function that reads/writes `m_background_syncing` would need to grab a lock, which adds a fair bit of complexity I think for little to no gain. I don't think an atomic would solve the problem here, but maybe I'm missing something. Can you expand on an example where an atomic would yield correct behavior where the current code doesn't?